drivers/ethernet/eth_mcux.c

```diff
@@ -887,8 +887,8 @@ static void generate_random_mac(u8_t *mac_addr)
 }
 #endif
 
-#if defined(CONFIG_ETH_MCUX_0_UNIQUE_MAC) || \
-    defined(CONFIG_ETH_MCUX_1_UNIQUE_MAC)
+#if !DT_INST_NODE_HAS_PROP(0, local_mac_address) || \
+    DT_HAS_NODE(DT_DRV_INST(1)) && !DT_INST_NODE_HAS_PROP(1, local_mac_address)
```

Comment:
Why does generating eth0 depend on instance 1?

Reply:
Because generate_eth1_unique_mac calls generate_eth0_unique_mac.  And we technically could have a case in which local-mac-address is set on inst0 and not on inst1.